go/vt/vttablet/tabletserver/schema/engine.go

```diff
@@ -424,10 +425,55 @@ func (se *Engine) reload(ctx context.Context, includeStats bool) error {
 		return err
 	}
 
+	innodbTablesStats := make(map[string]*Table)
+	if includeStats {
+		if innodbTableSizesQuery := conn.Conn.BaseShowInnodbTableSizes(); innodbTableSizesQuery != "" {
+			// Since the InnoDB table size query is available to us on this MySQL version, we should use it.
+			// We therefore don't want to query for table sizes in getTableData()
+			includeStats = false
+
+			innodbResults, err := conn.Conn.Exec(ctx, innodbTableSizesQuery, maxTableCount, false)
+			if err != nil {
+				return vterrors.Wrapf(err, "in Engine.reload(), reading innodb tables")
+			}
+			for _, row := range innodbResults.Rows {
+				innodbTableName := row[0].ToString() // In the form of encoded `schema/table`
+				fileSize, _ := row[1].ToCastUint64()
+				allocatedSize, _ := row[2].ToCastUint64()
+
+				if _, ok := innodbTablesStats[innodbTableName]; !ok {
+					innodbTablesStats[innodbTableName] = &Table{}
+				}
+				// There could be multiple appearances of the same table in the result set:
+				// A table that has FULLTEXT indexes will appear once for the table itself,
+				// with total size of row data, and once for the aggregates size of all
+				// FULLTEXT indexes. We aggregate the sizes of all appearances of the same table.
+				table := innodbTablesStats[innodbTableName]
+				table.FileSize += fileSize
+				table.AllocatedSize += allocatedSize
+
+				if strings.Contains(innodbTableName, "#p#") {
+					// innodbTableName is encoded any special characters are turned into some @0-f0-f0-f value.
+					// Therefore this "#p#" here is a clear indication that we are looking at a partitioned table.
```

Comment:
Nice! Confirmed that with the encoded names we don't have to worry about tables that actually have these special chars in them as this example is encoded as `te@0023p@0023st`:
```
mysql> create table `te#p#st` (id int);
Query OK, 0 rows affected (0.01 sec)

mysql> SELECT it.name, its.file_size as normal_tables_sum_file_size, its.allocated_size as normal_tables_sum_allocated_size FROM information_schema.innodb_tables it JOIN information_schema.innodb_tablespaces its ON (its.space = it.space) WHERE its.name LIKE CONCAT(database(), '/%') AND its.name NOT LIKE CONCAT(database(), '/fts_%') UNION ALL SELECT it.name, SUM(its.file_size) as hidden_tables_sum_file_size, SUM(its.allocated_size) as hidden_tables_sum_allocated_size FROM information_schema.innodb_tables it JOIN information_schema.innodb_tablespaces its ON (  its.name LIKE CONCAT(database(), '/fts_', CONVERT(LPAD(HEX(table_id), 16, '0') USING utf8mb3) COLLATE utf8mb3_general_ci, '_%') ) WHERE  its.name LIKE CONCAT(database(), '/fts_%') GROUP BY it.name;
+-----------------------------+-----------------------------+----------------------------------+
| name                        | normal_tables_sum_file_size | normal_tables_sum_allocated_size |
+-----------------------------+-----------------------------+----------------------------------+
| vt_commerce/product         |                      114688 |                           114688 |
| vt_commerce/customer        |                      114688 |                           114688 |
| vt_commerce/corder          |                      114688 |                           114688 |
| vt_commerce/te@0023p@0023st |                      114688 |                           114688 |
| vt_commerce/members#p#p0    |                      114688 |                           114688 |
| vt_commerce/members#p#p1    |                      114688 |                           114688 |
| vt_commerce/members#p#p2    |                      114688 |                           114688 |
| vt_commerce/members#p#p3    |                      114688 |                           114688 |
| vt_commerce/members#p#p4    |                      114688 |                           114688 |
| vt_commerce/members#p#p5    |                      114688 |                           114688 |
+-----------------------------+-----------------------------+----------------------------------+
10 rows in set, 1 warning (0.01 sec)
```